sockeye/inference.py

```diff
@@ -888,9 +888,9 @@ def _get_inference_input(self,
         """
         batch_size = len(trans_inputs)
         lengths = [len(inp) for inp in trans_inputs]
-        source_length = mx.nd.array(lengths, ctx=self.context, dtype='float32')  # shape: (batch_size,)
+        source_length = mx.nd.array(lengths, ctx=self.context, dtype=self.dtype)  # shape: (batch_size,)
         max_length = max(len(inp) for inp in trans_inputs)
-        source = mx.nd.zeros((batch_size, max_length, self.num_source_factors), ctx=self.context, dtype='float32')
+        source_npy = np.zeros((batch_size, max_length, self.num_source_factors), dtype=np.float32)
```

Comment:
What's the reason to switch to numpy here?

Reply:
"Gathered the input token ids into a numpy array on CPU before sending them all to the GPU at the beginning of beam search, rather than sending each batch element to the GPU one at a time"

Copying the input to the GPU at once is faster.

Reply:
sure, but the same could have been achieved via `ctx=mx.cpu()`, no? Anyway, not a big deal. We can always change this to MXNet if we need this in the future.